src/hooks/useSearchResults.ts

```diff
@@ -22,34 +22,38 @@ export const useSearchResults = () => {
     [resultData]
   );
 
-  return (currentQuery: string) => {
-    const currentResults = storeIndex
-      .search(currentQuery, { expand: true })
-      .map(({ ref }) => storeIndex.documentStore.getDoc(ref) as SearchResult);
-
-    const mapResult = (result: SearchResult) => {
-      if (result.tableOfContents) {
-        const tableArray = result.tableOfContents.split('\n');
-
-        const uniqueItems = new Set(
-          tableArray
-            .map(replaceDataTagFromString)
-            .filter(item => compareTwoStrings(currentQuery, item) > 0.3)
-        );
-
-        return [...uniqueItems].map(item => ({
-          title: result.title,
-          displayTitle: item.replace(/(`|\(.*\))/g, ''),
-          id: `${result.id}-${createSlug(item)}`,
-          slug: `${result.slug}#${createSlug(item)}`,
-          category: result.category,
-          wrapInCode: item.startsWith('`'),
-        }));
-      }
-
-      return result;
-    };
-
-    return currentResults.slice(0, 20).map(mapResult).flat().slice(0, 20);
-  };
+  const searchResults = useCallback(
```

Comment:
I'm not sure this memoization will work the way you intended. Afaik `useCallback` will memoize the function execution based on its dependencies. This means, yes, the function will not be "recreated" for every `storeIndex` change. But the `storeIndex` is already memorized.

This function will still execute on every render for every given `currentQuery` even if `currentQuery` is the same. In the end.

I'm just making an assumption, of course, we can check if this function is being invoked on every re-render for the same content by adding `debugger;` or a `console.debug`. I might be wrong, of course.

Reply:
`searchResults` function will be recreated only if  `storeIndex` changes. 

Previously, The problem was even if we don't type any search query or with same search query the `searchResults`  function was gettting executed if any re-render happens in the app.

I've verified this change by adding console.log [here](https://github.com/nodejs/nodejs.dev/blob/1cf55e70729af9ca1b2e1b2d63f8c65cc51f0563/src/components/CommonComponents/SearchBar/index.tsx#L42)

```jsx
const results = useMemo(() => {
  console.log('fired');
  return search(query);
}, [query, search]);
```
## Before this PR
https://user-images.githubusercontent.com/42532987/200134153-28cdcb7b-bd15-4586-9264-57fd6dbdf6bd.mp4

## After this PR
https://user-images.githubusercontent.com/42532987/200134132-2b2a5b19-c5c9-4a53-b792-62aff2b33d3e.mp4

In video before this PR, you can notice that `fired` console logged every time when I open a modal or click something which leads to re-render. 
After this PR, the `fired` console logged only if I change the `query` and not on other re-renders.

Reply:
Yea, I wasn't discussing if the memoization helps or not. I'm just saying that you're memoizing the index, and of course as there are 0 memoizations the code inside the hook gets executed in every render, which is honestly **fine** as it is not a complex logic.

The memorization helps tho. But it would be even better to have memorization based on your search query. But maybe that's a little bit toooo much.